TESTS/mbed_hal/watchdog/main.cpp

```diff
@@ -201,9 +224,6 @@ Case cases[] = {
         test_update_config,
         (utest::v1::case_teardown_handler_t) case_teardown_wdg_stop_or_reset),
 
-    // Do not set watchdog timeout shorter than 500 ms as it may cause the
-    // host-test-runner return 'TIMEOUT' instead of 'FAIL' / 'PASS' if watchdog
-    // performs reset during test suite teardown.
     Case("Init, 500 ms", (utest::v1::case_setup_handler_t) case_setup_sync_on_reset,
```

Comment:
The test still initializes the watchdog with a timeout of 500ms, which will fail for boards that don't support it.

Reply:
Ok, I'll update the default timeout value to `100 ms` for all watchdog tests and push the commit to this PR after I test that locally.

What about `tests-mbed_hal-watchdog_timing`? https://github.com/ARMmbed/mbed-os/blob/29d6dcae808355308849b7f77f5397b0007990fe/TESTS/mbed_hal/watchdog_timing/main.cpp#L136-L139 There is no point in running this test on devices with such a short timeout, especially on the CI, where communication delays aren't constant. Shall we add some macro and execute this test conditionally?

Reply:
Local tests pass. Update pushed.

Reply:
I don't think we should assume what timeouts are supported by a specific board.

Can the test check the max supported timeout before running the timing test and then conditionally run it if it's supported?